orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/persistence/DualExecutionRepository.kt

```diff
@@ -164,6 +164,14 @@ class DualExecutionRepository(
     return select(type, id).delete(type, id)
   }
 
+  override fun delete(type: ExecutionType, idsToDelete: MutableList<String>) {
+    // NOTE: Not a great implementation, but this method right now is only used on SqlExecutionRepository which has
+    // a performant implementation
```

Comment:
I don't follow - why can't we use `delete(type, idsToDelete)` here? Why do we have to iterate over each ID?

Reply:
we don't know which repos to perform the deletes on. I guess i could call:
```java
primary.delete(type, idsToDelete)
previous.delete(type, idsToDelete)
```

but depending on what primary and previous are (e.g. not SQL) then this is even less performant, probably?

Reply:
eh, i take it back, it's the same since there is a select somewhere no matter what